### PR TITLE
JDK-8319301: Static analysis warnings after JDK-8318016

### PR DESCRIPTION
**Pull request by tstuefe**
Sonar reports two new troubles in compileOracle.cpp:

1) "left operand of '-' is a garbage value", when seeing that `line` might not be initialized properly:

```
  char* end;
  if (!parse_integer<size_t>(line, &end, &s)) {
    jio_snprintf(errorbuf, buf_size, "MemLimit: invalid value");
  }
  bytes_read = (int)(end - line); // <---- here
```

2) "3rd function call argument is an uninitialized value", because there is an early `return true` from parseMemLimit that does not initialize `value`:

```
   // Special handling for memlimit
    bool success = (option == CompileCommand::MemLimit) && parseMemLimit(line, value, bytes_read, errorbuf, buf_size);
    if (!success) {
      // Is it a raw number?
      success = sscanf(line, "" INTX_FORMAT "%n", &value, &bytes_read) == 1;
    }
    if (success) {
      total_bytes_read += bytes_read;
      line += bytes_read;
      register_command(matcher, option, value); // <---- here
```

Patch fixes both instances and adds new regression test.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8319301](https://bugs.openjdk.org/browse/JDK-8319301): Static analysis warnings after JDK-8318016 (**Bug** - P3)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [70ca8327](https://git.openjdk.org/jdk/pull/16631/files/70ca8327c46c93fed3da2f37e5b5a890baa12093)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [024d6813](https://git.openjdk.org/jdk/pull/16631/files/024d681308e9eb8125bab8fda138e1089d703210)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16631/head:pull/16631` \
`$ git checkout pull/16631`

Update a local copy of the PR: \
`$ git checkout pull/16631` \
`$ git pull https://git.openjdk.org/jdk.git pull/16631/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16631`

View PR using the GUI difftool: \
`$ git pr show -t 16631`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16631.diff">https://git.openjdk.org/jdk/pull/16631.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16631#issuecomment-1809751352)

### CODE REVIEW

File: test/hotspot/jtreg/compiler/compilercontrol/commands/MemLimitTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8318671
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/compilercontrol/commands/MemLimitTest.java:27]: The bug number points to the wrong bug.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me otherwise.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by tstuefe** [comment]
Thanks @TobiHartmann and @vnkozlov . Will let the GHAs run through one more time after fixing the merge conflicts, then push if green.

**Pull request closed without merging**